client/my-sites/plans/current-plan/jetpack-checklist/index.tsx

```diff
@@ -115,18 +119,45 @@ class JetpackChecklist extends PureComponent< Props > {
 					completedTitle={ task.completedTitle }
 					description={ task.description }
 					duration={ task.duration }
+					firstIncomplete={ this.getFirstIncompleteTask( taskIds ) }
 					href={ task.getUrl( this.props.siteSlug, isComplete ) }
+					id={ taskId }
+					key={ taskId }
 					onClick={ this.handleTaskStart( {
 						taskId,
 						tourId: get( task, [ 'tourId' ] ),
 					} ) }
+					onTaskTitleClick={ this.onTaskTitleClick }
 					title={ task.title }
-					key={ taskId }
+					selectedTaskId={ this.state.selectedTaskId }
 				/>
 			);
 		} );
 	}
 
+	onTaskTitleClick = selectedTaskId => {
+		this.setState( { selectedTaskId } );
+		this.props.recordTracksEvent( 'calypso_checklist_task_expand', {
+			step_name: selectedTaskId,
+			product: 'Jetpack',
+		} );
+	};
+
+	getFirstIncompleteTask = taskStatuses => {
```

Comment:
This is some duplicated code in both the Wp.com checklist and Jetpack code. I'm not sure where to put it or how to code it in such a way that it becomes centralized. Any thoughts @ockham or @simison?

Reply:
Good thinking! If it's generically applicable to checklist, it should be moved to `components/checklist/`.